src/submodules/pathfinding/astar_continuous.jl

```diff
@@ -0,0 +1,154 @@
+    isempty(discrete_path) && return
+    cts_path = Path{D,Float64}()
+    for pos in discrete_path
+        push!(cts_path, pos ./ pathfinder.grid_dims .* model.space.extent .- half_cell_size)
+    end
+    last_pos = last(cts_path)
+    pop!(cts_path)
+    second_last_pos = isempty(cts_path) ? from : last(cts_path)
+    last_to_end = edistance(last_pos, to, model)
+    second_last_to_end = edistance(second_last_pos, to, model)
+    if last_to_end < second_last_to_end
+        push!(cts_path, last_pos)
+    end
+    last_to_end ≈ 0. || push!(cts_path, to)
+    return cts_path
+end
+
+"""
+    Pathfinding.set_target!(agent, target::NTuple{D,Int}, pathfinder)
+Calculate and store the shortest path to move the agent from its current position to
+`target` (a grid position e.g. `(1, 5)`) for using the provided `pathfinder`.
+
+Use this method in conjuction with [`move_along_route!`](@ref).
+"""
+function set_target!(
+    agent::A,
+    target::NTuple{D,Float64},
+    pathfinder::AStar{D},
+    model::ABM{<:ContinuousSpace{D},A},
+) where {D,A<:AbstractAgent}
+    path = find_continuous_path(pathfinder, agent.pos, target, model)
+    isnothing(path) && return
+    pathfinder.agent_paths[agent.id] = path
+end
+
+"""
+    Pathfinding.set_best_target!(agent, targets::Vector{NTuple{D,Int}}, pathfinder)
+
+Calculate and store the best path to move the agent from its current position to
+a chosen target position taken from `targets` for models using [`Pathfinding`](@ref).
+
+The `condition = :shortest` keyword retuns the shortest path which is shortest
+(allowing for the conditions of the models pathfinder) out of the possible target
+positions. Alternatively, the `:longest` path may also be requested.
+
+Returns the position of the chosen target.
+"""
+function set_best_target!(
+    agent::A,
+    targets::Vector{NTuple{D,Float64}},
+    pathfinder::AStar{D},
+    model::ABM{<:ContinuousSpace{D}};
+    condition::Symbol = :shortest,
+) where {D,A<:AbstractAgent}
+    @assert condition ∈ (:shortest, :longest)
+    compare = condition == :shortest ? (a, b) -> a < b : (a, b) -> a > b
+    best_path = Path{D,Float64}()
+    best_target = nothing
+    for target in targets
+        path = find_continuous_path(pathfinder, agent.pos, target, model)
+        isnothing(path) && continue
+        if isempty(best_path) || compare(length(path), length(best_path))
+            best_path = path
+            best_target = target
+        end
+    end
+
+    pathfinder.agent_paths[agent.id] = best_path
+    return best_target
+end
+
+"""
+    get_direction(from, to, space)
+Returns the direction vector from `from` to `to` taking into account periodicity of the space
+(for continuous space)
+"""
+# TODO: Dispatch on AStar's periodicity, since it's _technically_ possible for it to be different
+# from the space
+function get_direction(from::NTuple{D,Float64}, to::NTuple{D,Float64}, space::ContinuousSpace{D,true}) where {D}
```

Comment:
If this is to be a ContinuousSpace exclusive, may want to add a `get_direction(from, to, model) = get_direction(from, to, model.space)` wrapper.